File: LeetCode/April Challenge/23rd Day/Day_23.cpp
class Solution {
public:
    int rangeBitwiseAnd(int m, int n) {
        if (m==0)
            return 0;
        int a=log2(m);
        int b=log2(n);
        if(a!=b)
            return 0;
        else
        {
            long long int res=m;
            for(long long int i=m;i<=n;i++)
                res&=i;
            return res;
        }
    }
};

File: LeetCode/April Challenge/23rd Day/Day_23.py
import math
class Solution:
    def rangeBitwiseAnd(self, m: int, n: int) -> int:
        if m == 0:
            return 0
        top = int(math.log(n, 2))
        bottam = int(math.log(m, 2))
        if top != bottam:
            return 0
        
        res = m
        for i in range(m + 1, n + 1):
            res &= i
        return res
